contracts/modules/assessment/Claims.sol

```diff
@@ -222,65 +230,74 @@ contract Claims is IClaims, RegistryAware {
+    ) =  address(pool).call{value: CLAIM_DEPOSIT_IN_ETH}("");
     require(transferSucceeded, AssessmentDepositTransferToPoolFailed());
-  
+
     return claim;
   }
 
-  /// Redeems payouts for accepted claims
+  /// Redeems payouts and sends assessment deposit back for accepted claims
   ///
-  /// @dev Anyone can call this function, the payout always being transfered to the NFT owner.
-  /// When the tokens are transfered the assessment deposit is also sent back.
+  /// @dev Can be called by anyone, but the payout and deposit are always transferred to the current cover NFT owner.
   ///
   /// @param claimId  Claim identifier
   function redeemClaimPayout(uint claimId) external override whenNotPaused(PAUSE_CLAIMS_PAYOUT) {
-    Claim memory claim = _validateClaimStatus(claimId, IAssessment.AssessmentStatus.ACCEPTED);
+    (Claim memory claim, uint cooldownEnd) = _validateClaimStatus(claimId, IAssessment.AssessmentStatus.ACCEPTED);
+
+    require(block.timestamp < cooldownEnd + PAYOUT_REDEMPTION_PERIOD, RedemptionPeriodExpired());
+    require(!claim.payoutRedeemed, PayoutAlreadyRedeemed());
 
     _claims[claimId].payoutRedeemed = true;
+    _claims[claimId].depositRetrieved = true;
 
-    _ramm().updateTwap();
-    address payable coverOwner = payable(_cover().burnStake(
+    ramm.updateTwap();
+
+    address payable coverOwner = payable(cover.burnStake(
       claim.coverId,
       claim.amount
     ));
 
     // Send payout in cover asset
-    _pool().sendPayout(claim.coverAsset, coverOwner, claim.amount, CLAIM_DEPOSIT_IN_ETH);
+    pool.sendPayout(claim.coverAsset, coverOwner, claim.amount, CLAIM_DEPOSIT_IN_ETH);
 
     emit ClaimPayoutRedeemed(coverOwner, claim.amount, claimId, claim.coverId);
   }
 
-  function retriveDeposit(uint claimId) external override whenNotPaused(PAUSE_CLAIMS_PAYOUT) {
-    Claim memory claim =_validateClaimStatus(claimId, IAssessment.AssessmentStatus.DRAW);
-    
-    _claims[claimId].payoutRedeemed = true;
+  /// Allows the cover owner to retrieve their claim deposit if their claim is resolved as DRAW.
+  ///
+  /// @dev Can be called by anyone, but the claim deposit is always transferred to the current cover NFT owner.
+  ///
+  /// @param claimId The unique identifier of the claim for which the deposit is being retrieved.
+  function retrieveDeposit(uint claimId) external override whenNotPaused(PAUSE_CLAIMS_PAYOUT) {
+    (Claim memory claim, /* cooldownEnd */) = _validateClaimStatus(claimId, IAssessment.AssessmentStatus.DRAW);
+
+    require(!claim.depositRetrieved, DepositAlreadyRetrieved());
 
-    address payable coverOwner = payable(_coverNFT().ownerOf(claim.coverId));
+    _claims[claimId].depositRetrieved = true;
 
-    _pool().returnDeposit(coverOwner, CLAIM_DEPOSIT_IN_ETH);
+    address payable coverOwner = payable(coverNFT.ownerOf(claim.coverId));
 
-    emit ClaimDepositRetrived(claimId, coverOwner);
+    pool.returnDeposit(coverOwner, CLAIM_DEPOSIT_IN_ETH);
```

Comment:
just a note here, more of a "to keep in mind": if the deposit changes at any point, we'll have to make sure there are no open claims during the contract upgrade. the alternative would be to store the deposit amount which I'm not sure if is worth implementing.